src/main/java/ro/msg/learning/shop/strategy/LocationStrategy.java

```diff
@@ -4,11 +4,12 @@
 import ro.msg.learning.shop.dto.ProductQuantityDto;
 import ro.msg.learning.shop.dto.StockDto;
 
+import java.math.BigDecimal;
 import java.util.List;
 
 @Component
 public interface LocationStrategy {
 
-    List<StockDto> findLocation(List<ProductQuantityDto> products);
+    List<StockDto> findLocation(List<ProductQuantityDto> products, List<BigDecimal> distances);
```

Comment:
You should not have parameters which are not relevant for all strategies, you will not use the distances in Single Location strategy implementation for example.